### PR TITLE
Added the Clang-tidy support (ENABLE_CLANG_TIDY option for CMake).

### PR DESCRIPTION
**Pull request by monadbobo**

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  - key:             modernize-pass-by-value.IncludeStyle
+    value:           llvm
+  - key:             modernize-replace-auto-ptr.IncludeStyle
+    value:           llvm
+  - key:             modernize-use-nullptr.NullMacros
+    value:           'NULL'
+  - key:             readability-identifier-naming.EnumConstantCase
+    value:           CamelCase
+  - key:             readability-identifier-naming.EnumConstantPrefix
+    value:           k
```
Comment (laura-ding) [inline .clang-tidy:140]: add line

File: cmake/ClangTidy.cmake [hunk 2]
```diff
@@ -0,0 +1,15 @@
+option (ENABLE_CLANG_TIDY "Enable clang-tidy if present" OFF)
```
Comment (laura-ding) [inline cmake/ClangTidy.cmake:1]: add into `cmake/nebula/GeneralCMakeOptions.cmake`. All option are define in it.

**Review by laura-ding — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]
LGTM.

Shall we add clang-tidy checks in CI ?

**Comment by monadbobo** [comment]
> LGTM.
> 
> Shall we add clang-tidy checks in CI ?

I will add clang-tidy to CI in a follow-up PR.

**Comment by dutor** [comment]
> > LGTM.
> > Shall we add clang-tidy checks in CI ?
> 
> I will add clang-tidy to CI in a follow-up PR.

I don't think it's approriate to perform the clang tidy testing in a daily. It's more of a complementary tool to help us discover some less critical issues.

**Pull request merged**